.github/workflows/discussions-old-topics-marked-solved.py

```diff
@@ -0,0 +1,87 @@
+name: Mark Discussions as Solved
+on:
+  schedule:
+    # Schedule this action to run periodically (e.g., daily)
+    - cron: '0 0 * * *'
+
+jobs:
+  mark-as-solved:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        run: npm install @actions/github octokit/rest.js
+
+      - name: Find and Mark Discussions as Solved
+        run: |
+          # Import necessary modules
+          const { Octokit } = require("@octokit/rest");
+          const githubToken = process.env.GITHUB_TOKEN;
+          const octokit = new Octokit({ auth: githubToken });
+
+          // Define the team name to check against
+          const teamName = "Reviewers";
+
+          // Calculate one year ago from the current date
+          const oneYearAgo = new Date();
+          oneYearAgo.setFullYear(oneYearAgo.getFullYear() - 1);
+
+          // Fetch discussions in the repository
+          const repoOwner = "sofa-framework";
+          const repoName = "sofa";
+          const discussionsResponse = await octokit.rest.issues.list({
+            owner: repoOwner,
+            repo: repoName,
+            q: `is:discussion is:open`,
+          });
+
+          // Process discussions
+          for (const discussion of discussionsResponse.data.items) {
+            // Get the oldest reply
+            const repliesResponse = await octokit.rest.issues.listComments({
+              owner: repoOwner,
+              repo: repoName,
+              issue_number: discussion.number,
+            });
+
+            if (repliesResponse.data.length > 0) {
+              const oldestReply = repliesResponse.data[0];
```

Comment:
Wouldn't it be the least old ?
`const oldestReply = repliesResponse.data[repliesResponse.data.length-1];`

Reply:
And what if `repliesResponse.data.length == 0` ? a discussion, older than a year that hasn't been answered will certainly never be answered at all. Shouldn't we just close it ?